packages/riverpod_lint/README.md

```diff
@@ -643,6 +644,91 @@ switch (...) {
 }
 ```
 
+### incorrect_usage_of_ref_method
+
+Warn if the `ref.watch`/`ref.read`/`ref.listen`/`ref.listenManual` methods are used incorrectly.
+
+**Good**:
+
+```dart
+final fnProvider = Provider<int>((ref) {
+  ref.watch(provider);
+  ref.listen(provider, ...);
+  return 0;
+});
+
+class MyNotifier extends Notifier<int> {
+  int get _readGetter => ref.read(provider);
+
+  @override
+  int build() {
+    ref.watch(provider);
+    ref.listen(provider, ...);
+    return 0;
+  }
+
+  void someMethod() {
+    ref.read(provider);
+  }
+}
+
+void initState() {
+  ref.listenManual(provider, ...);
+}
+
+Widget build(ctx, ref) {
+  ref.watch(provider);
+  ref.listen(provider, ...);
+
+  return Button(
+    onPressed: () {
+      ref.read(provider);
+      ref.listenManual(provider, ...);
+    }
+  );
+}
+```
+
+**Bad**:
+
+```dart
+final fnProvider = Provider<int>((ref) {
+  ref.read(provider); // use watch instead
+  return 0;
+});
+
+class MyNotifier extends Notifier<int> {
+  int get _watchGetter => ref.watch(provider); // use read instead
```

Comment:
I don't recommend making such getters at all. They can introduce bugs implicitly, because the getter may be used in multiple places where some places want `read` and others want `watch`

Reply:
I see. I will remove it.